### DISCUSSION TITLE
Modal dialog please Help

### DISCUSSION THREAD

**Discussion by jcarnalich** [microsoft/WindowsAppSDK#2603] [General]
Hello, 
we are a company from software development. We have programs made in **MFC/C++** with thousands of customers.
We want to migrate our products from **MFC/C++** to **WinUI 3**. 

The main problem is that we use many modal dialogs. Many times from a modal dialog opens another modal dialog, and another... (see screenshot):

![example_mfc](https://user-images.githubusercontent.com/19662787/172792732-68bfc810-eef7-403a-b64d-f3ffdffd9105.png)

I have two questions:

**1° plans Windows to do this type of modal dialogs in the future?
2° Is it possible to make modal dialogs from other modal dialogs with the current version of winui ?** 

we have tried to do the following without success:

a) **Contentdialog/Flyout**. It's ok but you can't open another Contentdialog/Flyout from a Contentdialog/Flyout.
b) Do it manually with **UIElement.GotFocus** Event. 
c) **Windowing**. AppWindowPresenterKind.CompactOverlay. It's ok but you can't open another Window (modal with CompactOverlay) from a Window.
d) WinUI Extensions : **WinUIEx**. (It's based also in Windowing)  It's ok but you can't open another WinUIEx from one WinUIEx.
e) Do it manually with **Canvas**, ManipulationMode="TranslateX, TranslateY, TranslateInertia", Visibility.Collapsed/Visible, IsHitTestVisible?

Could you please help us?
Thanks in advance!
Best regards

**Comment by castorix** [comment]
A way is to use the Window class, as it is possible now to create multiple windows on the same thread
A test, with [EnableWindow ](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-enablewindow) to disable/enable parent windows :

        [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
        public static extern bool EnableWindow(IntPtr hWnd, bool bEnable);

![Modal_windows](https://user-images.githubusercontent.com/22345506/172849482-63b1e649-6363-45c7-b999-91a9885587b1.gif)

**Comment by jcarnalich** [comment]
Thank you very much! I have installed the latest version of SDK and your solution works now with **windowing**. 

It is a possible solution, but two questions please :

**1° are there WinUI-Elements that allow me to do modal dialogs inside a window?**
**2° does Microsoft plan to do modal dialogs?**

I have done something similar to a modal dialog with _Canvas_ und _ContentControl_ _(ManipulationStarted, ManipulationDelta, ManipulationCompleted, Tapped, PointerEntered, Visibility, IsHitTextVisible)_
but it is not a clean and good solution. (see screenshot)

![modal_dialog_canvas_one_window3](https://user-images.githubusercontent.com/19662787/173018021-de89fc04-d6a1-4e3e-99a6-ff1aaa5701b1.png)


Best regards
